.github/workflows/build-test.yml

```diff
@@ -0,0 +1,48 @@
+name: Build & Test
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+env:
+  DOCKER_ENV_FILE: ".github/workflows/docker.env"
+jobs:
+  build-test:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:12.2-alpine
+        env:
+          POSTGRES_DB: "development"
```

Comment:
Can we substitute these values from the environment/`docker.env` file as well, so changes (if ever needed) would only need to be applied to one place?

Reply:
Please no separate `docker.env` file. It's particulary bad as this format is not in YAML.

Reply:
The reason for the `docker.env` file is because the env set in the job does not apply to the container. Therefore the env for the created container need to be set by some workaround.
Regarding the reuse of variables, I tried to set the env vars in the YAML and use the something like `env > docker.env` as the env-file for docker, but then the `PATH` in the docker image is also overridden which leads to issues.

Reply:
Can you subsitute env vars for these values (e.g. `POSTGRES_DB: "${PGDATABASE]"`)?

Reply:
w/o further changes sadly no, because `PGDATABASE` is only available in the docker container and this is not supported by actions https://github.community/t/env-vars-are-not-expanded-causing-dry-violations/16550.  An option would be to set everything in env in the yaml and reuse it with `POSTGRES_DB: ${{ env.PGDATABASE }}` (https://github.community/t/how-to-use-env-context/16975/2) but the doc does not specify if it is also available for services.

But then the remaining problem is to set the env in the docker container.

Reply:
For now I recommend that we keep the github flow including its inside its *separate* `.github/workflows/docker.env` file.